lib/system_parser.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class SystemParser
+  DfRecord = Struct.new(:unix_device, :optional_name, :size_gib, :avail_gib)
+
+  def self.get_device_mount_points_from_lsblk_json(json_input, device_name)
+    data = JSON.parse(json_input)
+    excluded_mounts = ["[SWAP]", "/boot", nil]
+
+    mount_points = {}
+
+    data["blockdevices"].each do |device|
+      next unless device_name == device["name"]
+
+      get_mounts = lambda do |dev|
+        dev["mountpoints"]&.each do |mount|
+          mount_points[device["name"]] ||= []
+          mount_points[device["name"]] << mount unless excluded_mounts.include?(mount)
```

Comment:
```suggestion
          (mount_points[device["name"]] ||= []) << mount unless excluded_mounts.include?(mount)
```

Alternatively, switch to having the unless around both lines.  The goal would be to not to have an empty array for an excluded mount.

Reply:
done thanks